From: Helmut Grohne <helmut@subdivi.de>
Subject: Do not run host arch executables during cross build
Last-Update: 2014-09-16

When cross building autogen, one has to install autogen for the build
architecture, because it uses itself during build.

Index: autogen-5.18.3/configure.ac
===================================================================
--- autogen-5.18.3.orig/configure.ac	2014-05-17 16:49:39.000000000 +0200
+++ autogen-5.18.3/configure.ac	2014-09-16 23:31:38.603484754 +0200
@@ -167,9 +167,15 @@
 AGnam=autogen${ac_exeext}
 GDnam=getdefs${ac_exeext}
 CLnam=columns${ac_exeext}
+if test "$cross_compiling" = "yes"; then
+AGexe=${AGnam}
+GDexe=${GDnam}
+CLexe=${CLnam}
+else
 AGexe=${ag_top_builddir}/agen5/${AGnam}
 GDexe=${ag_top_builddir}/getdefs/${GDnam}
 CLexe=${ag_top_builddir}/columns/${CLnam}
+fi
 M4_SRC=`cd $srcdir/config ; echo [a-z]*.m4`
 ENABLE_STATIC=${enable_static}
 config_end_time=`date +%s 2>/dev/null`
Index: autogen-5.18.3/xml2ag/Makefile.am
===================================================================
--- autogen-5.18.3.orig/xml2ag/Makefile.am	2014-01-10 20:20:57.000000000 +0100
+++ autogen-5.18.3/xml2ag/Makefile.am	2014-09-17 11:31:21.755480980 +0200
@@ -34,8 +34,6 @@
 DOCFILES        = invoke-$(TARG).texi invoke-$(TARG).menu $(TARG).1
 DISTCLEANFILES  = $(DOCFILES) $(nodist_xml2ag_SOURCES) stamp-*
 AM_CPPFLAGS     = @INCLIST@ $(LIBXML2_CFLAGS)
-AGexe           = $(top_builddir)/agen5/autogen
-CLexe           = $(top_builddir)/columns/columns
 BOOTENV     = top_srcdir="$(top_srcdir)" top_builddir="$(top_builddir)" \
 	PATH=`cd ../columns;pwd`:"$$PATH" CLexe="$(CLexe)"
 
